README.md

```diff
@@ -295,6 +295,81 @@ curl http://127.0.0.1:8080/such_path
 > such_path
 ```
 
+## Events
+
+The router emits three events - `handlestart`, `layer`, and `handleend` - as it processes requests.
+
+### 1. `handlestart`
+
+This event is emitted when the router starts to process a request.
+
+Example:
+
+```js
+router.on('handlestart', function (req) {
+  req.id = Math.random().toString(36).slice(2)
+  console.log('HANDLESTART:', req.id)
+})
+```
+
+### 2. `layer`
+
+This event is emitted when a route layer is matched.
+
+Layers are matched till one of them sends a response back to the client.
+
+Example:
+
+```js
+router.on('layer', function (req, layer) {
+  console.log(layer)
+})
+```
+
+### 3. `handleend`
+
+This event is emitted when the router has finished processing a request and sent a response.
+
+Example:
+
+```js
+router.on('handleend', function (req) {
+  console.log('HANDLEEND:', req.id)
+})
+```
+
+Here is a complete example of using router events in an Express 5 app.
+
+```js
+var express = require('express')
```

Comment:
Can the example be a real example instead of just a bunch of console log lines? I.e. demonstrate the use case people will use this for. People love to copy paste, and this is not something they would do that on.